lib/orderbook/MatchesProcessor.ts

```diff
@@ -1,38 +1,90 @@
 import Logger from '../Logger';
 import { OrderMatch } from '../types/matchingEngine';
-import { isPeerOrder } from '../types/orders';
+import { isPeerOrder, StampedPeerOrder, StampedOwnOrder } from '../types/orders';
 import RaidenClient from '../raidenclient/RaidenClient';
 import Pool from '../p2p/Pool';
+import * as packets from '../p2p/packets';
+import uuidv1 from 'uuid/v1';
+import { SwapDeal } from './SwapDeals';
+import { SwapDealRole } from '../types/enums';
 
 class MatchesProcessor {
-  private buffer: any[];
-
-  constructor(private logger: Logger, private pool?: Pool, private raidenClient?: RaidenClient) {
-    this.buffer = [];
-  }
-
-  public add(match: OrderMatch) {
-    this.buffer.push(match);
+  constructor(private logger: Logger, private pool?: Pool, private raidenClient?: RaidenClient) { }
 
+  public process = (match: OrderMatch) => {
     if (isPeerOrder(match.maker)) {
-      this.notifyPeer(match);
-    } else if (isPeerOrder(match.taker)) {
       this.executeSwap(match);
+    } else if (isPeerOrder(match.taker)) {
```

Comment:
This condition is no longer relevant - taker is always `ownOrder`.

We have only 2 conditions: either maker is `peerOrder` (non-internal match), or it's not (internal match).

Reply:
Yeah I thought this too but wasn't sure to drop it. Basically this is relevant to the, as far as I understand, still unresolved question of what to do when you receive an order from a peer that *should* match immediately. But I'm pretty sure that shouldn't be considered a "match" in any case, so I'll cut out this code and make sure we have an issue for it to discuss.